client/state/preferences/selectors.js

```diff
@@ -36,6 +36,26 @@ export function getPreference( state, key ) {
 	);
 }
 
+export function getMediaScalePreference( state, key, isMobile ) {
+	const SCALE_TOUCH_GRID = 0.323;
+	const mediaScale = get(
+		find(
+			[
+				state.preferences?.localValues,
+				state.preferences?.remoteValues,
+				DEFAULT_PREFERENCE_VALUES,
+			],
+			( source ) => has( source, key )
+		),
+		key,
+		null
+	);
+
+	if ( ( isMobile && mediaScale !== 1 ) || ( ! isMobile && mediaScale > SCALE_TOUCH_GRID ) ) {
+		return SCALE_TOUCH_GRID;
+	}
```

Comment:
This condition deserves a comment that explains it. And maybe also being divided into two conditions:
1. On mobile view, allow only 1 and 0.323 scales.
2. On non-mobile view, don't allow the 1 scale: always show at least 3 columns.

And if the `getMediaScalePreference` selector ends up being called only at one place, let's inline it in that module. The selector is too specialized to be in the general-purpose `state/preferences`  module.